src/cloud/services/WorkflowService.ts

```diff
@@ -829,26 +831,128 @@ const registerTransportReject = async (
   }
 };
 
+/**
+ * Check if the recicling center has the balance to pay the transaction
+ * @param containersPayment
+ * @param recyclingCenter
+ * @param user
+ */
+const checkBalanceTransaction = async ({
+  containersPayment,
+  recyclingCenter,
+  user,
+}: {
+  containersPayment: { container: string; total: number; payment: number }[];
+  recyclingCenter: Parse.Object;
+  user: Parse.User;
+}): Promise<void> => {
+  const totalPayment = containersPayment
+    .map((item) => item.payment)
+    .reduce((partialSum, a) => partialSum + a, 0);
+
+  // Find the containers by the code
+  const retributionUpdate = await Promise.all(
+    containersPayment.map((element) =>
+      ContainerService.findContainerByCode({
+        code: element.container,
+      }),
+    ),
+  );
+
+  // find all the transaction where the containers are used
+  const historyWithTransaction = await Promise.all(
+    retributionUpdate.map(
+      (element) =>
+        element &&
+        TransactionService.generateHistoryTransactionFromContainer([element.id], user, true).then(
+          (hTransaction) => ({
+            containerId: element.id,
+            containerName: element.get('code'),
+            history: hTransaction,
+          }),
+        ),
+    ),
+  );
+
+  // find tranports transaction from the history
+  const transactionTransport = historyWithTransaction.map(
+    (transaction) =>
+      transaction &&
+      transaction.history.filter((element) => element.type === TRANSACTIONS_TYPES.TRANSPORT),
+  );
+
+  let ammountTransport = 0;
+  transactionTransport.forEach((transaction) => {
+    if (transaction) {
+      transaction.forEach((historyTransaction) => {
+        ammountTransport += historyTransaction.retribution.estimated;
+      });
+    }
+  });
+
+  // Check the balance of the recycling center
+  const { balance } = await getBalance(recyclingCenter.get('walletId'));
+  if (balance <= totalPayment + ammountTransport) {
+    throw new Error(`The Recycling Center has not enough money, the balance is ${balance} jyc`);
```

Comment:
hide the balance. Show only "insufficient funds"

Reply:
done